script/support/sample_move.rb

```diff
@@ -0,0 +1,296 @@
+# frozen_string_literal: true
+
+class StudySample < ApplicationRecord
+end
+
+# added sample class as comments are not commentable at present 21/3/2019
+class Sample
+  include Commentable
+end
+
+class Asset
+  has_many :submitted_assets
+  has_many :orders, through: :submitted_assets, as: :asset
+end
+
+def update_fluidigm_plates(fluidigm_plates)
+  fluidigm_plates.each do |plate_id|
+    puts "Updating Fluidigm plate: #{plate_id}"
+    plate = Plate.find_by id: plate_id
+
+    # rubocop:disable Rails/SkipsModelValidations
+    # correct behaviour
+    plate.touch
+    plate.save!
+    message = Messenger.find_by(target_id: plate_id)
+    message.resend
+  end
+end
+
+def find_asset_groups(sample_names)
+  hash = Hash.new { |hsh, key| hsh[key] = [] }
+  sample_names.each do |sample_name|
+    sample = Sample.find_by(name: sample_name)
+    asset_groups = sample.asset_groups
+    unless asset_groups.empty?
+      asset_groups.each { |ag| hash[ag.name] << sample_name unless Order.find_by(asset_group_id: ag.id).nil? }
+    end
+  end
+  hash
+end
+
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Metrics/MethodLength
+# rubocop:disable Metrics/AbcSize
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Layout/LineLength
+def find_whole_and_split_asset_groups(asset_group_sample_hash, sample_names)
+  # determine which orders are split if any
+  whole_asset_groups = []
+  split_asset_groups_hash = Hash.new { |hsh, key| hsh[key] = [] }
+  asset_group_sample_hash.each do |ag_name, samples|
+    # puts "ag => #{ag_name}"
+    asset_group = AssetGroup.find_by(name: ag_name)
+    order = Order.find_by(asset_group_id: asset_group.id)
+    aliquot_count = asset_group.assets.map(&:aliquots).flatten.size
+    if aliquot_count == samples.size
+      puts "#{order.study_id} #{order.submission_id} #{order.submission.orders.size} #{ag_name} - #{aliquot_count}/#{samples.size} :: All assets to be moved"
+
+      whole_asset_groups << asset_group
+      # if assetgroup asset == MultiplexedLibraryTube then only the aliquots can be updated
```

Comment:
Does this comment relate to the `elsif` below?  It's either in the wrong position or incorrectly indented if so.